### PR TITLE
Spec testing for enumerable methods using Rspec

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
- Wrote tests for each method using Rspec.
- describe Enumerable group.
- Used describe block and it block
- Created variables using let

### CODE REVIEW

File: spec/enum_spec.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    end
+  end
+
+  describe '#my_count' do
+    it 'count array elements' do
+      expect(array.my_count).to eql(4)
+    end
+    it 'total number against the passed argument' do
+      expect(array.my_count).to eql(4)
+    end
```
Comment (kubilaycaglayan) [inline spec/enum_spec.rb]: :bulb: There is no passed argument here.

File: spec/enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    it 'total number for condition true' do
+      expect(array.my_count { |e| e > 1 }).to eql(3)
+    end
+  end
+
+  describe '#my_map' do
+    it 'returns a new array of elements for which condition true' do
+      expect(array.my_map { |e| e + 1 }).to eql([2, 3, 4, 5])
+    end
```
Comment (kubilaycaglayan) [inline spec/enum_spec.rb]: :bulb: This definition `returns a new array of elements for which condition true` seems to belongs to the "select" method but not map. Please fix it.

File: spec/enum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  let(:range) { (1..4) }
+  describe '#my_each' do
+    it 'return each element of an array' do
+      expect(array.my_each { |element| }).to eql(array.my_each { |element| })
+    end
+    it 'return each element of an hash' do
+      expect(hash.my_each { |element| }).to eql(hash.my_each { |element| })
+    end
+    it 'return each element of a range' do
+      expect(range.my_each { |element| }).to eql(range.my_each { |element| })
```
Comment (kubilaycaglayan) [inline spec/enum_spec.rb]: :bulb:  What we are doing here is to check the condition with itself. Instead, we should pass the result into the matcher. The same applies to the my_each_with_index tests.

File: spec/enum_spec.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    end
+  end
+
+  describe '#my_all?' do
+    it 'return true if all of the elements are true for condition' do
+      expect(array.my_all? { |element| element >= 1 }).to eql(true)
+    end
+    it 'return false if all of the elements are false for condition' do
+      expect(array.my_all? { |element| element > 2 }).to eql(false)
+    end
```
Comment (kubilaycaglayan) [inline spec/enum_spec.rb:66]: :bulb:  It statement is `return false if all of the elements are false for condition`. 
Maybe you wanted to say `returns false if the block ever returns false or nil`
Please check the documentation to see&understand how they are verbalizing it => https://ruby-doc.org/core-2.6.5/Enumerable.html

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team :wave: 
Status: Changes Required :recycle: 

Great job so far! :muscle: :blush: 
All your tests are passing :heavy_check_mark: 

Let's cover some important details about the project to make it perfect :dart:  :arrow_down: 

### Readme

- [x] The project has a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

:bulb: I see that you have mentioned about tests. Please add a section about how to run tests, also.

### GitHub Actions + Linter Checks

:bulb: Currently, your GitHub actions for linter checks are not working. If you think you have properly configured them please try to turn GitHub Actions off/on again from repository settings. 

### Tests

- [x] All methods implemented in the Enumerables project are tested ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_enumerables.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/testing-your-ruby-code))

:bulb: Please add the `multiply_els` method into your file again and create a test for it too.

- [x]  Fix `my_each` and `my_each_with_index` tests. Currently, the matchers are equals to the tests itself. 

:bulb: Please check the inline comments below :arrow_down: 

- [x]  Fix `my_all` definition.

:bulb: Please check the inline comments below :arrow_down: 

- [x]  Fix `my_count`.

:bulb: Please check the inline comments below :arrow_down: 

- [x]  Fix `my_map` test definition.

:bulb: Please check the inline comments below :arrow_down: 

###  [OPTIONAL] 

- [x] Using `require_relative` for files in your directory instead of `require` is a ruby convention.

### That's it, team! After doing these changes, you can request another review. :blush: 

Thanks for your time and efforts :+1: 
With all my respect... :pray:

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [ ] Let's write tests for the following methods when a block is given.
- my_each
- my_each_with_index
- my_inject
- my_count
- my_any
- my_all
- my_select
- my_none
- [ ] Let's write test for when a Class is passed as an argument.
- [ ] Also when regex is passed as an argument .

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**